src/diffusers/models/autoencoders/autoencoder_kl_mochi.py

```diff
@@ -0,0 +1,705 @@
+            return self.tiled_decode(z, return_dict=return_dict)
+
+        dec = self.decoder(z)
+
+        if not return_dict:
+            return (dec,)
+
+        return DecoderOutput(sample=dec)
+
+    @apply_forward_hook
+    def decode(self, z: torch.Tensor, return_dict: bool = True) -> Union[DecoderOutput, torch.Tensor]:
+        """
+        Decode a batch of images.
+
+        Args:
+            z (`torch.Tensor`): Input batch of latent vectors.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether to return a [`~models.vae.DecoderOutput`] instead of a plain tuple.
+
+        Returns:
+            [`~models.vae.DecoderOutput`] or `tuple`:
+                If return_dict is True, a [`~models.vae.DecoderOutput`] is returned, otherwise a plain `tuple` is
+                returned.
+        """
+        if self.use_slicing and z.shape[0] > 1:
+            decoded_slices = [self._decode(z_slice).sample for z_slice in z.split(1)]
+            decoded = torch.cat(decoded_slices)
+        else:
+            decoded = self._decode(z).sample
+
+        if not return_dict:
+            return (decoded,)
+
+        return DecoderOutput(sample=decoded)
+
+    def blend_v(self, a: torch.Tensor, b: torch.Tensor, blend_extent: int) -> torch.Tensor:
+        blend_extent = min(a.shape[3], b.shape[3], blend_extent)
+        for y in range(blend_extent):
+            b[:, :, :, y, :] = a[:, :, :, -blend_extent + y, :] * (1 - y / blend_extent) + b[:, :, :, y, :] * (
+                y / blend_extent
+            )
+        return b
+
+    def blend_h(self, a: torch.Tensor, b: torch.Tensor, blend_extent: int) -> torch.Tensor:
+        blend_extent = min(a.shape[4], b.shape[4], blend_extent)
+        for x in range(blend_extent):
+            b[:, :, :, :, x] = a[:, :, :, :, -blend_extent + x] * (1 - x / blend_extent) + b[:, :, :, :, x] * (
+                x / blend_extent
+            )
+        return b
+
+    def tiled_decode(self, z: torch.Tensor, return_dict: bool = True) -> Union[DecoderOutput, torch.Tensor]:
+        r"""
+        Decode a batch of images using a tiled decoder.
+
+        Args:
+            z (`torch.Tensor`): Input batch of latent vectors.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~models.vae.DecoderOutput`] instead of a plain tuple.
+
+        Returns:
+            [`~models.vae.DecoderOutput`] or `tuple`:
+                If return_dict is True, a [`~models.vae.DecoderOutput`] is returned, otherwise a plain `tuple` is
+                returned.
+        """
+
+        batch_size, num_channels, num_frames, height, width = z.shape
+
+        tile_latent_min_height = self.tile_sample_min_height // self.spatial_compression_ratio
+        tile_latent_min_width = self.tile_sample_min_width // self.spatial_compression_ratio
+        tile_latent_stride_height = self.tile_sample_stride_height // self.spatial_compression_ratio
```

Comment:
Made some changes to the tiling logic we copied from CogVideoX VAE tiling, because all the overlap related variables did not make sense naming-wise. Better to call it stride for clarity. Okay to propagate these changes to CogVideoX and Allegro as well @DN6? Can open follow-up PRs next week after testing.

Another thing to note is that frame-wise tiling support isn't added yet because I couldn't figure out how exactly the time upscaling is happening - it does not seem similar to CogVideoX. Due to this, we can only decode a maximum of about 97 frames from my testing on 24 GB when model CPU offloading is enabled. I'll spend some time next week trying to figure out how temporal tiling can be done here

Reply:
@a-r-r-o-w Some OOM error occurs in VAE when I use 143 frames in A100 80GB. Do you want to implement a parallel version of VAE? That could be the answer to OOM. CPU offload is not enough.

Reply:
Hi @foreverpiano. Since Diffusers is tailored to single GPU inference, we do not plan on adding a parallel version of the VAE. Maybe in the future multi-GPU inference would be considered but at the moment this will not be possible.

I'm working on frame-wise decoding approach for the VAE here, and if it works out, we should be able to decode a lot more frames than the current limitation

Reply:
That makes sense. 
The VAE is the cause to OOM. Enable tiling and other method will be great.